services/httpd/msgp_emitter.go

```diff
@@ -0,0 +1,160 @@
+package httpd
+
+import (
+	"io"
+	"net/http"
+
+	"github.com/influxdata/influxdb/influxql"
+	"github.com/tinylib/msgp/msgp"
+)
+
+type messagePackEncoder struct {
+	Epoch     string
+	ChunkSize int
+}
+
+func (e *messagePackEncoder) ContentType() string {
+	return "application/x-msgpack"
+}
+
+func (e *messagePackEncoder) Encode(w io.Writer, header ResponseHeader, results <-chan *influxql.ResultSet) {
+	var convertToEpoch func(row *influxql.Row)
+	if e.Epoch != "" {
+		convertToEpoch = epochConverter(e.Epoch)
+	}
+	values := make([][]interface{}, 0, e.ChunkSize)
+
+	enc := msgp.NewWriter(w)
+	enc.WriteMapHeader(1)
+	enc.WriteString("results")
+	enc.WriteInt(header.Results)
```

Comment:
There are 2 ways the client knows when to stop reading a sequence of objects. Either the client knows how many objects to read, or it receives something which tells it to stop.

But my point is that having an indicator when to stop reading works for everything. Having a count that tells the client how many to read only works for some things (when you know what that count is). Thus for consistency (and less confusion), the former works better.